### PR TITLE
chore: normalize npm metatdata

### PR DESCRIPTION
**Pull request by NetaNir**
clean up the data we get from the registry using [normalize-registry-metadata](https://github.com/npm/normalize-registry-metadata), provided by npm

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: src/backend/discovery/discovery.lambda.ts [hunk 1]
```diff
@@ -23,13 +26,11 @@ export const STAGED_KEY_PREFIX = 'staged/';
...
  *  - a message will be sent to an sqs queue
- * Currently we don't handle the function execution timeout, and accept that the last batch processed might be processed again,
- * relying on the idempotency on the consumer side.
  * npm registry API docs: https://github.com/npm/registry/blob/master/docs/REGISTRY-API.md
  * @param context a Lambda execution context
  */
-export async function handler(event: ScheduledEvent, context: Context) {
-  console.log(`Event: ${JSON.stringify(event, null, 2)}`);
+export async function handler( context: Context) {
+  // console.log(`Event: ${JSON.stringify(event, null, 2)}`);
```
Comment (RomainMuller) [inline src/backend/discovery/discovery.lambda.ts:33]: Why?

File: src/backend/discovery/discovery.lambda.ts [hunk 2]
```diff
@@ -264,6 +265,7 @@ function getRelevantVersionInfos(changes: readonly Change[]): readonly UpdatedVe
       continue;
     }
 
+    normalizeNPMMetadata(change.doc);
```
Comment (RomainMuller) [inline src/backend/discovery/discovery.lambda.ts:268]: My understanding is this returns the cleaned document... But you're actually not using the result of the function?

File: src/backend/discovery/discovery.lambda.ts [hunk 3]
```diff
@@ -23,13 +26,11 @@ export const STAGED_KEY_PREFIX = 'staged/';
...
  *  - the package version tarball will be copied from the npm registry to a stating bucket.
  *  - a message will be sent to an sqs queue
- * Currently we don't handle the function execution timeout, and accept that the last batch processed might be processed again,
- * relying on the idempotency on the consumer side.
  * npm registry API docs: https://github.com/npm/registry/blob/master/docs/REGISTRY-API.md
  * @param context a Lambda execution context
  */
-export async function handler(event: ScheduledEvent, context: Context) {
-  console.log(`Event: ${JSON.stringify(event, null, 2)}`);
+export async function handler( context: Context) {
```
Comment (RomainMuller) [inline src/backend/discovery/discovery.lambda.ts:32]: The context is the second argument to Lambda invocations... This isn't going to work...

File: src/backend/discovery/discovery.lambda.ts [hunk 4]
```diff
@@ -372,3 +374,26 @@ interface Change {
...
+  },
+  succeed: () => {
+  },
+  fail: () => {
+  },
+  done: () => {
+  },
+};
+
+handler(context).catch((error) => {console.log(error);});
```
Comment (RomainMuller) [inline src/backend/discovery/discovery.lambda.ts:399]: Leftover test code?

**Review by RomainMuller — CHANGES_REQUESTED** [review]

File: src/backend/discovery/discovery.lambda.ts (hunk 1 above)
Comment (NetaNir) [reply src/backend/discovery/discovery.lambda.ts:33]: dammit, debugging mistake :/

File: src/backend/discovery/discovery.lambda.ts (hunk 4 above)
Comment (NetaNir) [reply src/backend/discovery/discovery.lambda.ts:399]: yeah I pushed the wrong commit, let me fix it up. The only change is the normalize line

File: src/backend/discovery/discovery.lambda.ts (hunk 3 above)
Comment (NetaNir) [reply src/backend/discovery/discovery.lambda.ts:32]: I pushed the wrong commit :/ fixing

**Pull request closed without merging**